File: tests/langbench/wc.py
#!/usr/bin/python
import os
import sys

def wordsplit(line):
    list = []
    word = ""
    for c in line:
	if c.isspace():
	    if len(word) > 0:
		list.append(word)
	    word = ""
	else:
	    word += c
    if len(word) > 0:
	list.append(word)
    return list

def main():
    n = 0
    for a in sys.argv[1:]:
    	f = open(a)
    	for line in f:
        	words = wordsplit(line)
		n += len(words)
    	f.close()
    print "%d\n" % n
   
if __name__ == "__main__":
    main()

File: tests/langbench/wc.rb
def wordsplit(line)
    list = []
    word = ""
    line.split('').each do |c|
	if c =~ /\s/
	    if word.length > 0
		list << word
	    end
	    word = ""
	else
	    word += c
	end
    end
    if word.length > 0
	list << word
    end
    return list
end

n = 0
while line = gets()
    words = wordsplit(line)
    n += words.length
end
printf("%d\n", n)
